File: bmi3/app/src/main/java/com/project/bmi2/Activity5.java
package com.project.bmi2;

import android.graphics.Color;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.jjoe64.graphview.GraphView;
import com.jjoe64.graphview.GridLabelRenderer;
import com.jjoe64.graphview.LegendRenderer;
import com.jjoe64.graphview.series.DataPoint;
import com.jjoe64.graphview.series.LineGraphSeries;
import com.jjoe64.graphview.helper.StaticLabelsFormatter;

public class Activity5 extends AppCompatActivity {
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_5);
        GraphView graph = (GraphView) findViewById(R.id.graph);
        StaticLabelsFormatter staticLabelsFormatter = new StaticLabelsFormatter(graph);
        LineGraphSeries<DataPoint> series = new LineGraphSeries<DataPoint>(new DataPoint[]{
                new DataPoint(0, 22),
                new DataPoint(1, 23),
                new DataPoint(2, 24),
                new DataPoint(3, 25),

        });
        graph.addSeries(series);

        staticLabelsFormatter.setHorizontalLabels(new String[]{"01-01-2023", "01-02-2023", "01-03-2023"});
        graph.getGridLabelRenderer().setLabelFormatter(staticLabelsFormatter);
        series.setColor(Color.rgb(0, 80, 100));
        series.setDrawDataPoints(true);
        series.setDataPointsRadius(5);
        series.setThickness(5);
        graph.addSeries(series);
        graph.setTitle("BMI progress");
        graph.setTitleTextSize(50);
        graph.setTitleColor(Color.RED);
        graph.getLegendRenderer().setVisible(true);
        graph.getLegendRenderer().setAlign(LegendRenderer.LegendAlign.TOP);
        GridLabelRenderer gridLabel = graph.getGridLabelRenderer();
        gridLabel.setHorizontalAxisTitle("X time");
        gridLabel.setVerticalAxisTitle("Y BMI");

    }

}

File: bmi_kotlin/app/src/main/java/com/project/bmi2/Activity5.kt
package com.project.bmi2

import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.jjoe64.graphview.GraphView
import com.jjoe64.graphview.LegendRenderer
import com.jjoe64.graphview.helper.StaticLabelsFormatter
import com.jjoe64.graphview.series.DataPoint
import com.jjoe64.graphview.series.LineGraphSeries

class Activity5 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_5)
        val graph = findViewById<View>(R.id.graph) as GraphView
        val staticLabelsFormatter = StaticLabelsFormatter(graph)
        val series = LineGraphSeries(arrayOf(
                DataPoint(0.0, 22.0),
                DataPoint(1.0, 23.0),
                DataPoint(2.0, 24.0),
                DataPoint(3.0, 25.0)))
        graph.addSeries(series)
        staticLabelsFormatter.setHorizontalLabels(arrayOf("01-01-2023", "01-02-2023", "01-03-2023"))
        graph.gridLabelRenderer.labelFormatter = staticLabelsFormatter
        series.color = Color.rgb(0, 80, 100)
        series.isDrawDataPoints = true
        series.dataPointsRadius = 5f
        series.thickness = 5
        graph.addSeries(series)
        graph.title = "BMI progress"
        graph.titleTextSize = 50f
        graph.titleColor = Color.RED
        graph.legendRenderer.isVisible = true
        graph.legendRenderer.align = LegendRenderer.LegendAlign.TOP
        val gridLabel = graph.gridLabelRenderer
        gridLabel.horizontalAxisTitle = "X time"
        gridLabel.verticalAxisTitle = "Y BMI"
    }
}